fix : removed dead provider with reason

dushyantbhati2 opened the pull request:
@tomkins
fixes #12054 
I have implemented a script named check_providers.py, which tests the functionality of various oEmbed providers. This script generates a list of non-functional providers, detailing the specific reasons each provider was deemed non-functional.

Subsequently, I have updated the oembed_providers.py file, removing the non-functional providers from the all_providers list.

lb- commented:
@dushyantbhati2 thank you for this PR. 

Can I suggest a different tact for this. 

1. For now, let's not add the check script, this appears to be very bespoke and it's probably not needed in the core code for now. Maybe it's worth a separate PR or as a CI check in the future. 
2. It looks like this removal is going to impact existing services and would likely be a breaking change. Instead, for all services that are still active (e.g. Spotify), prepare a standalone PR that fixes that one specific service, updating to the new URLs. 
3. For all services that have shut down, update this PR to only remove those, update the description with a reference to each being removed and when it was discontinued. E.g. Rdio 2015 https://en.m.wikipedia.org/wiki/Rdio

This way we can get momentum on this PR as a focused 'clean up' of discontinued services. Maybe let's have a threshold of ~5 years.

Awesome to see this cleaned up, hopefully you're ok to work through the feedback.

dushyantbhati2 commented:
I have updated the URLs for providers that have changed their endpoints and removed those that have been discontinued. Each non-functional provider in the list now includes a reason for its removal and each updated provider is mentioned in updated_urls_provider. For reference, the updates are based on information from [oEmbed's official provider list](https://oembed.com/providers.json).

lb- commented on wagtail/embeds/oembed_providers.py:
```diff
@@ -770,6 +763,140 @@
+
+
+'''
+updated_urls_provider=[
```
Thanks @dushyantbhati2 - these notes can just go on the PR as a comment or description update. No need to put this in the code.

lb- commented on wagtail/embeds/oembed_providers.py:
```diff
@@ -199,12 +199,12 @@
+        r"^http://channel-islands\.geograph\.org/.*$",
+        r"^http://channel-islands\.geographs\.org/.*$",
+        r"^http://.*\.channel\.geographs\.org/.*$"
+    ]
```
I this these geographs ones would still need https right?
Also, double check you have run the formatting, it looks like the nesting is out here. 

See https://docs.wagtail.org/en/stable/contributing/python_guidelines.html#python-coding-guidelines

dushyantbhati2 replied:
Thank you for bringing this to my attention. The correct URL should use "https," and I will ensure this is corrected in the next commit.

lb- commented on wagtail/embeds/oembed_providers.py:
```diff
@@ -522,8 +522,8 @@
-        r"^https?://open\.spotify\.com/.+$",
-        r"^https?://spoti\.fi/.+$",
+        r"^https://open\.spotify\.com/.*$",
+        r"^spotify:.*$"
```
I'm nor sure we would want the Spotify scheme here, I would assume most URLs users copy would not be this format. 

Do you have an example of docs link where this would be the case?

dushyantbhati2 replied:
@lb- Thank you for pointing that out. You are correct; Spotify scheme is not widely used by most users. Therefore, I have removed it. I appreciate your help in correcting my mistake.

lb- reviewed (changes requested):
A few more bits of feedback. Thank you @dushyantbhati2 

Please review the CI checks for issues also.

dushyantbhati2 commented:
@lb- Thank you very much for your guidance and assistance with this PR. Is there anything else I need to address? Please feel free to let me know if you have any additional feedback or requests.

RealOrangeOne commented on wagtail/embeds/oembed_providers.py:
```diff
@@ -199,11 +197,11 @@
-        r"^https?://channel-islands\.geograph\.org/.+$",
-        r"^https?://channel-islands\.geographs\.org/.+$",
-        r"^https?://(?:[-\w]+\.)?channel\.geographs\.org/.+$",
+        r"^https?://.*\.geograph\.org\.gg/.*$",
```
Question: Could you explain this change and why it's needed? This pattern looks much broader than the previous.

gasman commented:
Completed in #12537.
